components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/ODataController.java

```diff
@@ -66,13 +65,15 @@ private JPAODataClaimsProvider createClaims(final Token token, String tenantID)
         }
 
         if (tenantID == null || tenantID.isEmpty()) {
-            logger.warn("Could not determine tenant claim");
-            return claims;
+            if (token.applicationTenantId != null && !token.applicationTenantId.isEmpty()) {
+                logger.warn("Application tenant ID is provided through header and tenant ID is empty");
+                tenantID = "";
```

Comment:
Placeholder comment: after the follow up PR, this if will have to be reworked/adapted because even if we have `applicationTenantId` we still can have non empty tenantID returned.